types/address/hash.go

```diff
@@ -67,8 +67,7 @@ func Module(moduleName string, key []byte) []byte {
 // unsafeStrToByteArray uses unsafe to convert string into byte array. Returned array
 // cannot be altered after this functions is called
 func unsafeStrToByteArray(s string) []byte {
-	sh := *(*reflect.SliceHeader)(unsafe.Pointer(&s)) // nolint:govet
-	sh.Cap = sh.Len
-	bs := *(*[]byte)(unsafe.Pointer(&sh)) // nolint:govet
-	return bs
+	var buf = *(*[]byte)(unsafe.Pointer(&s))
+	(*reflect.SliceHeader)(unsafe.Pointer(&buf)).Cap = len(s)
```

Comment:
This can work but the form that directly follows the standard way of conversion will be clearer for other Go developers in the future aka
```go
        sh := (*reflect.SliceHeader)(unsafe.Pointer(&s))
        sh.Cap = sh.Len
        return *(*[]byte)(unsafe.Pointer(sh))
```

Reply:
I discussed it in https://github.com/cosmos/cosmos-sdk/pull/8674#issuecomment-785003585
I think it's not correct because we are modifying `sh.Cap` - since in your proposal we are doing a pointer cast of a smaller structure, we will write on someones' else memory.